lnst/Recipes/ENRT/BaseSRIOVNetnsTcRecipe.py

```diff
@@ -108,19 +104,15 @@ def test_wide_configuration(self):
 
         # create virtual functions
         for host in [host1, host2]:
-            host.run(f"devlink dev eswitch set pci/{host.eth0.bus_info} mode switchdev")
-            time.sleep(2)
-            host.run(f"echo 1 > /sys/class/net/{host.eth0.name}/device/sriov_numvfs")
-            time.sleep(3)
-
-            vf_ifname = dict(ifname=f"{host.eth0.name}{self.params.vf_suffix}")
-            host.map_device("vf_eth0", vf_ifname)
+            host.eth0.eswitch_mode = "switchdev"
+            host.eth0.create_vfs(1)
+            vf_device = host.eth0.vf_devices[0]
+            vf_rep_device = host.eth0.vf_rep_devices[0]
```

Comment:
What about using dataclass to hold vf device + vf representor device (Optional)

```
from dataclasses import dataclass
from typing import Optional

@dataclass
class SRIOVDevices:
    virtual_function: Device
    virtual_function_representor: Optional[Device] = None

vf_only = SRIOVDevices(virtual_function=vf)
device_pair = SRIOVDevices(virtual_function=vf, virtual_function_representor=vf_repr)
```

This object be returned in a list if there are multiple VF pairs / VFs created.
Also this allows us to implement or move methods that can be used for checks/creation of VFs, depending how you implementing it.

Reply:
the location makes sense to me, it is right now used just for the `BaseSRIOVNetnsTcRecipe` hierarchy... we can move it once it becomes more generically used. In case we want to make some generalization adjustments for that...